pkg/image/nbdkit.go

```diff
@@ -161,6 +169,24 @@ func NewNbdkitVddk(nbdkitPidFile, socket, server, username, password, thumbprint
 	return n, nil
 }
 
+func writePasswordFile(password string) (string, error) {
+	f, err := os.CreateTemp("", "password")
+	if err != nil {
+		return nil, err
+	}
```

Comment:
You need `defer f.Close()` here so that the file is closed if the `Write` fails. Iirc even with the `defer` added, you can keep the explicit `f.Close()` at the end of the function as this will allow you to catch `Close()` errors.